traitsui/tests/editors/test_liststr_editor_selection.py

```diff
@@ -74,21 +92,539 @@ class ListStrEditorWithSelectedIndex(HasTraits):
+            model.value = value
+
+        with store_exceptions_on_all_threads():
+            gui, editor = self.setup_gui(model, get_view())
+
+            set_selected_single(editor, 0)
+            gui.process_events()
+            # Sanity check
+            self.assertEqual(editor.selected_index, 0)
+            self.assertEqual(editor.selected, "one")
+
+            editor.callx(change_value, model, ["two", "one"])
+            gui.process_events()
+
+            # Nothing is updated
+            self.assertEqual(get_selected_indices(editor), [0])
+            self.assertEqual(editor.selected_index, 0)
+            self.assertEqual(editor.selected, "one")
+
+    @skip_if_not_qt4  # wx editor doesn't have a `setx` method
+    def test_list_str_editor_setx(self):
+        with store_exceptions_on_all_threads():
+            gui, editor = self.setup_gui(ListStrModel(), get_view())
+
+            set_selected_single(editor, 0)
+            gui.process_events()
+            # Sanity check
+            self.assertEqual(editor.selected_index, 0)
+            self.assertEqual(editor.selected, "one")
+
+            editor.setx(selected="two")
+            gui.process_events()
+
+            # Specified attribute is modified
+            self.assertEqual(editor.selected, "two")
+            # But nothing else is updated
+            # FIXME issue enthought/traitsui#867
+            with self.assertRaises(AssertionError):
+                self.assertEqual(get_selected_indices(editor), [0])
+                self.assertEqual(editor.selected_index, 0)
+            self.assertEqual(get_selected_indices(editor), [1])
+            self.assertEqual(editor.selected_index, 1)
+
+    def test_list_str_editor_horizontal_lines(self):
+        # Smoke test for painting horizontal lines
+        with store_exceptions_on_all_threads():
+            self.setup_gui(ListStrModel(), get_view(horizontal_lines=True))
+
+    def test_list_str_editor_title(self):
+        # Smoke test for adding a title
+        with store_exceptions_on_all_threads():
+            self.setup_gui(ListStrModel(), get_view(title="testing"))
+
+    @skip_if_not_wx
+    def test_list_str_editor_menu(self):
+        class ListStrModelRightClick(HasTraits):
+            value = List(["one", "two", "three"])
+            right_clicked = Str()
+            right_clicked_index = Int()
+
+        model = ListStrModelRightClick()
+        view = get_view(
+            right_clicked="object.right_clicked",
+            right_clicked_index="object.right_clicked_index",
+        )
+
+        with store_exceptions_on_all_threads():
+            gui, editor = self.setup_gui(model, view)
+
+            self.assertEqual(model.right_clicked, "")
+            self.assertEqual(model.right_clicked_index, 0)
+
+            right_click_item(editor.control, 1)
+            gui.process_events()
+
+            self.assertEqual(model.right_clicked, "two")
+            self.assertEqual(model.right_clicked_index, 1)
```

Comment:
I added a comment in `right_click_item` function Qt implementation. I managed to programatically launch the context menu but couldn't figure out how to close it. And since this is not high priority, I let it go. Also, the test as is wouldn't work for Qt anyways because of #868 point 5

Reply:
I see. Thank you. If you could reference #868  in the test case, or in the relevant test(s), that would be really helpful when other developers read these test in the future outside the context of a PR and wonder the same questions.